scripts/hyper-pod-template.json

```diff
@@ -14,7 +14,7 @@
         "--",
         "/bin/sh",
         "-c",
-        "while ! [ -f /opt/sd/setup.sh ]; do sleep 1; done; /opt/sd/setup.sh && (/opt/sd/launch --api-uri API_URI --store-uri STORE_URI --emitter /sd/emitter --build-timeout SD_BUILD_TIMEOUT BUILD_ID & /opt/sd/logservice --emitter /sd/emitter --api-uri API_URI --store-uri STORE_URI --build BUILD_ID & wait $(jobs -p))"
+        "while ! [ -f /opt/sd/setup.sh ]; do sleep 1; done; /opt/sd/setup.sh && (SD_TOKEN=`/opt/sd/launch --only-fetch-token --token \"BUILD_TOKEN\" --api-uri API_URI --store-uri STORE_URI --emitter /sd/emitter --build-timeout SD_BUILD_TIMEOUT BUILD_ID` && (/opt/sd/launch --api-uri API_URI --store-uri STORE_URI --emitter /sd/emitter --build-timeout SD_BUILD_TIMEOUT BUILD_ID & /opt/sd/logservice --emitter /sd/emitter --api-uri API_URI --store-uri STORE_URI --build BUILD_ID & wait $(jobs -p)))"
```

Comment:
I am wondering should we just provide this in a wrapper script. The whole 
```
(SD_TOKEN=`/opt/sd/launch --only-fetch-token --token \"BUILD_TOKEN\" --api-uri API_URI --store-uri STORE_URI --emitter /sd/emitter --build-timeout SD_BUILD_TIMEOUT BUILD_ID` && (/opt/sd/launch --api-uri API_URI --store-uri STORE_URI --emitter /sd/emitter --build-timeout SD_BUILD_TIMEOUT BUILD_ID & /opt/sd/logservice --emitter /sd/emitter --api-uri API_URI --store-uri STORE_URI --build BUILD_ID & wait $(jobs -p))
``` 

part in `/opt/sd/run.sh`

I am seeing that we are repeating this in all executors, and can be error prone. So why not wrap this in a script which launcher exposes and controls ? So all executors have to do is invoke `/opt/sd/run.sh` in addition to any extra scripts they have to run

Reply:
Input to the script `/opt/sd/run.sh` will be TEMP_TOKEN, API_URI,  STORE_URI, SD_BUILD_TIMEOUT & BUILD_ID

I'm just worried that we are putting logic of fetching token in multiple executors. A better approach might be to let launcher handle it along with invocation of logservice.

Reply:
I think it should better to provide a wrapper script. I'll fix launcher and executors to use such wrapper script for the first step. To fix to let launcher handle logic of fetching token along with invocation of logservice is second step.